internals/models.py

```diff
@@ -972,113 +974,135 @@ def put(self, notify=True, **kwargs):
     return key
 
   # Metadata.
-  created = db.DateTimeProperty(auto_now_add=True)
-  updated = db.DateTimeProperty(auto_now=True)
-  updated_by = db.UserProperty()
-  created_by = db.UserProperty()
+  created = ndb.DateTimeProperty(auto_now_add=True)
+  updated = ndb.DateTimeProperty(auto_now=True)
+  updated_by = ndb.UserProperty()
+  created_by = ndb.UserProperty()
 
   # General info.
-  category = db.IntegerProperty(required=True)
-  name = db.StringProperty(required=True)
-  feature_type = db.IntegerProperty(default=FEATURE_TYPE_INCUBATE_ID)
-  intent_stage = db.IntegerProperty(default=0)
-  summary = db.StringProperty(required=True, multiline=True)
-  unlisted = db.BooleanProperty(default=False)
+  category = ndb.IntegerProperty(required=True)
+  name = ndb.StringProperty(required=True)
+  feature_type = ndb.IntegerProperty(default=FEATURE_TYPE_INCUBATE_ID)
+  intent_stage = ndb.IntegerProperty(default=0)
+  summary = ndb.StringProperty(required=True)
+  unlisted = ndb.BooleanProperty(default=False)
   # TODO(jrobbins): Add an entry_state enum to track app-specific lifecycle
   # info for a feature entry as distinct from process-specific stage.
-  deleted = db.BooleanProperty(default=False)
-  motivation = db.StringProperty(multiline=True)
-  star_count = db.IntegerProperty(default=0)
-  search_tags = db.StringListProperty()
-  comments = db.StringProperty(multiline=True)
-  owner = db.ListProperty(db.Email)
-  footprint = db.IntegerProperty()  # Deprecated
+  deleted = ndb.BooleanProperty(default=False)
+  motivation = ndb.StringProperty()
+  star_count = ndb.IntegerProperty(default=0)
+  search_tags = ndb.StringProperty(repeated=True)
+  comments = ndb.StringProperty()
+  # owner = db.ListProperty(db.Email)
+  owner = ndb.StringProperty(repeated=True)
+  footprint = ndb.IntegerProperty()  # Deprecated
 
   # Tracability to intent discussion threads
-  intent_to_implement_url = db.LinkProperty()
-  intent_to_ship_url = db.LinkProperty()
-  ready_for_trial_url = db.LinkProperty()
-  intent_to_experiment_url = db.LinkProperty()
-  i2e_lgtms = db.ListProperty(db.Email)  # Currently, only one is needed.
-  i2s_lgtms = db.ListProperty(db.Email)
+  # intent_to_implement_url = db.LinkProperty()
+  intent_to_implement_url = ndb.StringProperty()
```

Comment:
`db.LinkProperty()` is deprecated. The replacement is `ndb.StringProperty()`  if  URLs are shorted than 500 characters. For longer URLs, `ndb.TextProperty()` is supposed to be used. Since I was unaware, I have replaced all instances of `db.LinkProperty()` with `ndb.StringProperty()`. If URLs are supposed to be longer than 500 characters, please let me know.

Reply:
It is 500 characters. As this [link](https://cloud.google.com/appengine/docs/standard/python/ndb/db_to_ndb) says,

> ndb.StringProperty() Has a max size of 500. For longer URLs, use ndb.TextProperty()

Reply:
@jrobbins, may we mark this conversation as resolved ?